### ISSUE TITLE
Product Text Input Box - Required

### ISSUE THREAD

**Issue by james7342** [HotcakesCommerce/hotcakes-commerce-core#452]
## Is your feature request related to a problem?
**Please describe.**
I have a name badge product that I want to require customer to provide their name before adding to cart. There is not an option to make a Text Input or HTML input "required" option in admin like you can for Drop-down, Radio button, check box.

## Describe the solution you'd like
In admin when creating a text or html input provide a option to select required.

## Describe alternatives you've considered
Thought about trying to set this in a view but it could be different depending on product. Tried to think of trying to use another product option as a way to trigger the input box to be required but haven't figured out a way to make that work.

**Comment by WillStrohl** [comment]
You are correct that those fields do not include the ability to require them at this time.  

With your help and others, we might be able to gather enough information to add this as an official feature.  To that end, can you please provide more information about how you're intending to use those fields with the required option, now and in the future?  Any detail is helpful.  :)  

In the meantime, if you wish to add this ability today yourself without waiting for and doing an upgrade, you still can.  However, it will require a bit of work for you.  This ability is outlined in the documentation article below.  

https://hotcakescommerce.zendesk.com/hc/en-us/articles/205426285-Save-Custom-Fields-During-Add-to-Cart-Checkout-and-Place-Order

**Comment by james7342** [comment]
The only use case I have at the moment is the example I provided, but could be applied to other products someone might want to sell. In my case it's a name badge product where the buyer would be required to input there name prior to adding to cart. Example: I sell a custom name badge or manufacture custom name badges. You order a name badge from my online store. I cannot create or manufacture the badge correctly unless you provide the name you wanted engraved, painted etc. Making the text box required for "Name" prior to adding to cart would force the buyer to provide the required details. 

Other scenario's could be any product that requires some sort of customization. Thinking Etsy style products where a buyer would want to add or customized text. It's not a static option, like someone selecting a size of shirt.
